packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -439,189 +276,30 @@ describe('NetworkController', () => {
                 },
               },
             ]);
-            createMetamaskProviderMock.mockReturnValue(fakeMetamaskProvider);
+            const fakeNetworkClient = buildFakeNetworkClient(fakeProvider);
+            createNetworkClientMock.mockReturnValue(fakeNetworkClient);
 
             await controller.initializeProvider();
 
-            expect(createMetamaskProviderMock).toHaveBeenCalledWith({
-              chainId: undefined,
-              engineParams: { pollingInterval: 12000 },
-              nickname: undefined,
+            expect(createNetworkClientMock).toHaveBeenCalledWith({
               rpcUrl: 'http://localhost:8545',
-              ticker: undefined,
+              chainId: toHex(1337),
+              type: NetworkClientType.Custom,
             });
             const { provider } = controller.getProviderAndBlockTracker();
+            assert(provider, 'Block tracker is somehow unset');
```

Comment:
Because `provider` is now properly typed, there's a chance that it could be undefined. We don't have a good way to guarantee this when calling `getProviderAndBlockTracker` (neither will consumers), so we have to check for it in tests.